chainforge/react-server/src/backend/query.ts

```diff
@@ -205,24 +208,29 @@ export class PromptPipeline {
 
         num_queries_sent += 1;
 
-        if (max_req > 1) {                
-          // Call the LLM asynchronously to generate a response, sending off
-          // requests in batches of size 'max_req' separated by seconds 'wait_secs' to avoid hitting rate limit
-          tasks.push(this._prompt_llm(llm, prompt, n, temperature, 
-                                      cached_resp, 
-                                      cached_resp_idx,
-                                      num_queries_sent, 
-                                      max_req, 
-                                      wait_secs, 
-                                      llm_params,
-                                      chat_history));
-        } else {
-          // Block. Await + yield a single LLM call.
-          let result = await this._prompt_llm(llm, prompt, n, temperature, 
-                                              cached_resp, cached_resp_idx, 
-                                              undefined, undefined, undefined, 
-                                              llm_params, chat_history);
-          yield this.collect_LLM_response(result, llm, responses);
+        try {
+          if (max_req > 1) {                
+            // Call the LLM asynchronously to generate a response, sending off
+            // requests in batches of size 'max_req' separated by seconds 'wait_secs' to avoid hitting rate limit
+            tasks.push(this._prompt_llm(llm, prompt, n, temperature, 
+                                        cached_resp, 
+                                        cached_resp_idx,
+                                        num_queries_sent, 
+                                        max_req, 
+                                        wait_secs, 
+                                        llm_params,
+                                        chat_history,
+                                        check_stop_condition));
+          } else {
+            // Block. Await + yield a single LLM call.
+            let result = await this._prompt_llm(llm, prompt, n, temperature, 
+                                                cached_resp, cached_resp_idx, 
+                                                undefined, undefined, undefined, 
+                                                llm_params, chat_history, check_stop_condition);
+            yield this.collect_LLM_response(result, llm, responses);
+          }
+        } catch (e) {
+          if (e instanceof UserForcedPrematureExit) { throw e; }
```

Comment:
This check is unnecessary because we want the error raised in prompt_llm to bubble up regardless.